class/class6.1/src/MessageDispatcher.cpp

```diff
@@ -0,0 +1,37 @@
+#include "MessageDispatcher.hpp"
+
+void MessageDispatcher::subscribeToGreenMessage(const GreenMessageHandler& handler) {
+    std::lock_guard<std::mutex> lock(mtx);
+    greenMessageHandlers.push_back(handler);
+}
+
+void MessageDispatcher::subscribeToBlueMessage(const BlueMessageHandler& handler) {
+    std::lock_guard<std::mutex> lock(mtx);
+    blueMessageHandlers.push_back(handler);
+}
+
+void MessageDispatcher::subscribeToOrangeMessage(const OrangeMessageHandler& handler) {
+    std::lock_guard<std::mutex> lock(mtx);
+    orangeMessageHandlers.push_back(handler);
+}
+
+void MessageDispatcher::publishGreenMessage(const GreenMessage& message) {
+    std::lock_guard<std::mutex> lock(mtx);
+    for (const auto& handler : greenMessageHandlers) {
+        handler(message);
```

Comment:
publish method is expected to be instant while in this case we will wat for all handlers to be completed therefore thread of publisher will hang for that time. Also you will lock a mutex for unpredictable time. It is better to use message queue and a separate thread to perform real dispatching